src/Cooked/MockChain/Balancing.hs

```diff
@@ -1,409 +1,326 @@
-        insRef <- txSkelReferenceInputUtxosPl skel
-        return $ ins <> insRef <> collateralUtxos
-      managedValidators <- txSkelInputValidators skel
-      theParams <- getParams
-      case estimateTxSkelFee theParams managedData managedTxOuts managedValidators attemptedSkel fee collateralIns of
-        -- necessary to capture script failure for failed cases
-        Left err@MCEValidationError {} -> throwError err
-        Left err -> throwError $ MCECalcFee err
-        Right newFee
-          | newFee == fee -> do
-              -- Debug.Trace.traceM "Reached fixpoint:"
-              -- Debug.Trace.traceM $ "- fee = " <> show fee
-              -- Debug.Trace.traceM $ "- skeleton = " <> show (attemptedSkel {_txSkelFee = fee})
-              pure (attemptedSkel, fee) -- reached fixpoint
-          | n == 0 -> do
-              -- Debug.Trace.traceM $ "Max iteration reached: newFee = " <> show newFee
-              pure (attemptedSkel, max newFee fee) -- maximum number of iterations
-          | otherwise -> do
-              -- Debug.Trace.traceM $ "New iteration: newfee = " <> show newFee
-              calcFee (n - 1) newFee skel collateralUtxos
+  return (txSkelBal, fee, adjustedCollateralIns, returnCollateralWallet)
 
--- | This funcion is essentially a copy of
--- https://github.com/input-output-hk/plutus-apps/blob/d4255f05477fd8477ee9673e850ebb9ebb8c9657/plutus-ledger/src/Ledger/Fee.hs#L19
-estimateTxSkelFee ::
-  Emulator.Params ->
-  Map Api.DatumHash Api.Datum ->
-  Map Api.TxOutRef Api.TxOut ->
-  Map Script.ValidatorHash (Script.Versioned Script.Validator) ->
-  TxSkel ->
-  Fee ->
-  Set Api.TxOutRef ->
-  Either MockChainError Fee
-estimateTxSkelFee params managedData managedTxOuts managedValidators skel fees collateralIns = do
-  txBodyContent <-
-    left MCEGenerationError $
-      generateBodyContent fees collateralIns params managedData managedTxOuts managedValidators skel
-  let nkeys = Cardano.estimateTransactionKeyWitnessCount txBodyContent
-  txBody <-
-    left (MCEGenerationError . TxBodyError "Error creating body when estimating fees") $
-      Cardano.createAndValidateTransactionBody Cardano.ShelleyBasedEraConway txBodyContent
-  case Cardano.evaluateTransactionFee Cardano.ShelleyBasedEraConway (Emulator.pEmulatorPParams params) txBody nkeys 0 of
-    Emulator.Coin fee -> pure $ Fee fee
+-- | This computes the minimum and maximum possible fee a transaction can cost
+-- based on the current protocol parameters
+getMinAndMaxFee :: (MonadBlockChainBalancing m) => m (Fee, Fee)
+getMinAndMaxFee = do
+  -- Default parameters in case they are not present. It is unclear when/if this
+  -- could actually happen though.
+  let defMaxTxExecutionUnits =
+        Cardano.ExecutionUnits {executionSteps = 10_000_000_000, executionMemory = 14_000_000}
+      defExecutionUnitPrices =
+        Cardano.ExecutionUnitPrices {priceExecutionSteps = 721 Rat.% 10_000_000, priceExecutionMemory = 577 Rat.% 10_000}
```

Comment:
Where do all these values come from? IMO there should be an explanation or link in the comment